src/base.c

```diff
@@ -25,15 +25,17 @@ base_map(extent_hooks_t *extent_hooks, unsigned ind, size_t size) {
 	if (extent_hooks == &extent_hooks_default) {
 		addr = extent_alloc_mmap(NULL, size, PAGE, &zero, &commit);
 	} else {
+		pre_reentrancy(tsdn_tsd(tsdn));
```

Comment:
Let's add an assert then?

I think this convinces me that we indeed shouldn't allow setting extent hooks on the shared arenas (unless we do so from the very beginning).